bs_constraints/include/bs_constraints/visual/inversedepth_reprojection_functor.h

```diff
@@ -0,0 +1,108 @@
+   */
+  InverseDepthReprojectionFunctor(const Eigen::Matrix2d& information_matrix,
+                                  const Eigen::Vector2d& pixel_measurement,
+                                  const Eigen::Matrix3d& intrinsic_matrix,
+                                  const Eigen::Matrix4d& T_cam_baselink,
+                                  const Eigen::Vector3d& bearing)
+      : information_matrix_(information_matrix),
+        pixel_measurement_(pixel_measurement),
+        intrinsic_matrix_(intrinsic_matrix),
+        T_cam_baselink_(T_cam_baselink),
+        bearing_(bearing) {}
+
+  /**
+   * @brief Construct a cost function instance for an inverse depth reprojection
+   * An inverse depth reprojection cost function is defined by two poses: the
+   * anchor pose of the inverse depth landmark, the measurement pose and the
+   * inverse depth landmark itself. The anchor pose is used to determine the
+   * euclidean location of the landmark in the world frame, which then gets
+   * projected into the measurement frame and the normal reprojection is
+   * computed.
+   *
+   * This transformation can be done by finding the relative transform between
+   * anchor and measurement frame and constructing a projection matrix:
+   * P = [K]*[R|t], where [R|t] is the 3x4 matrix representing the
+   * transformation from the anchor frame to the measurement frame.
+   */
+  template <typename T>
+  bool operator()(const T* const o_WORLD_BASELINKa,
+                  const T* const p_WORLD_BASELINKa,
+                  const T* const o_WORLD_BASELINKm,
+                  const T* const p_WORLD_BASELINKm,
+                  const T* const inverse_depth, T* residual) const {
+    // get extrinsic
+    Eigen::Matrix<T, 4, 4> T_CAM_BASELINK = T_cam_baselink_.cast<T>();
+
+    // get anchor pose as 4x4 matrix
+    Eigen::Matrix<T, 4, 4> T_WORLD_BASELINKa =
+        bs_constraints::OrientationAndPositionToTransformationMatrix(
+            o_WORLD_BASELINKa, p_WORLD_BASELINKa);
+
+    // get measurement pose as 4x4 matrix
+    Eigen::Matrix<T, 4, 4> T_WORLD_BASELINKm =
+        bs_constraints::OrientationAndPositionToTransformationMatrix(
+            o_WORLD_BASELINKm, p_WORLD_BASELINKm);
+
+    // get relative pose between anchor and measurement
+    Eigen::Matrix<T, 4, 4> T_CAMERAm_CAMERAa =
+        T_CAM_BASELINK * bs_constraints::InvertTransform(T_WORLD_BASELINKm) *
+        T_WORLD_BASELINKa * bs_constraints::InvertTransform(T_CAM_BASELINK);
+
+    // create projection matrix
+    const Eigen::Matrix<T, 3, 4> projection_matrix =
+        intrinsic_matrix_.cast<T>() * T_CAMERAm_CAMERAa.block(0, 0, 3, 4);
+
+    // compute the inverse depth and bearing vector (mx, my, 1, 1/Z)
+    Eigen::Matrix<T, 4, 1> bearing_and_inversedepth;
+    bearing_and_inversedepth << bearing_.cast<T>(), inverse_depth;
+
+    // project into measurement image
+    Eigen::Matrix<T, 2, 1> reproj =
+        (projection_matrix * bearing_and_inversedepth).hnormalized();
```

Comment:
I don't understand the math here. Where did you get this math from? I tried to google inverse depth parameterization and came to [this](https://www.doc.ic.ac.uk/~ajd/Publications/civera_etal_tro2008.pdf) paper. and I'm not sure how it's similar to what you have here.